### DISCUSSION TITLE
Forcing BCE for DiceCELoss when overlapping labels predictions

### DISCUSSION THREAD

**Discussion by SebGoll** [Project-MONAI/MONAI#8624] [Q&A]
Hello,

I have encountered multiple projects in which I need to output multiple labels for a single voxel. In this case, I'd like to use the `DiceCELoss`, but forcing it to use the binary cross entropy loss instead of the cross entropy one (since each channel predicted is either on or off independently of the other channels). 
In the current state of `DiceCELoss`, it automatically choose to use either ce/bce based on the number of channels:
https://github.com/Project-MONAI/MONAI/blob/e72145ce852e1b59e2dbf6d57d3360979c2f9407/monai/losses/dice.py#L804

I wanted to see if there is an officially supported way to achieve the behavior described above before contributing.

Thank you

**Comment by denis-samatov** [comment]
Yes — for **overlapping labels**, each output channel is an independent binary target, so the CE branch of `DiceCELoss` is not the loss you want. In MONAI 1.6.0, [`DiceCELoss.forward`](https://github.com/Project-MONAI/MONAI/blob/1.6.0/monai/losses/dice.py) chooses `CrossEntropyLoss` whenever the prediction has more than one channel; `sigmoid=True` changes only its Dice component. There is no `force_bce` switch for multi-channel input.

You can compose the two supported losses explicitly:

```python
import torch
from monai.losses import DiceLoss

dice = DiceLoss(
    sigmoid=True,
    softmax=False,
    to_onehot_y=False,
    include_background=True,
)
bce = torch.nn.BCEWithLogitsLoss()

def multilabel_loss(logits: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
    # Both tensors: [B, C, *spatial]; target is multi-hot (0/1), overlaps allowed.
    target = target.to(dtype=logits.dtype)
    return dice(logits, target) + bce(logits, target)
```

Pass **raw logits** to both terms: `DiceLoss` applies its own sigmoid, and `BCEWithLogitsLoss` incorporates the sigmoid numerically stably. Do not use `softmax`, `argmax`, or `to_onehot_y=True` for independent overlapping labels. If all your channels are foreground labels, keep `include_background=True` so channel 0 is not dropped; change it only if channel 0 really represents a separate background channel. For inference, use `torch.sigmoid(logits)` and threshold each channel independently.

I checked this with a synthetic voxel belonging to two labels on MONAI 1.6.0 / PyTorch 2.9.0: the composed loss was finite and gradients propagated. That verifies the API path, not performance on a real segmentation dataset. If you need imbalance handling, add `pos_weight` to `BCEWithLogitsLoss` based on your training data rather than assuming the default weighting is optimal.
